### ISSUE TITLE
PDMTS does not preserve modification timestamp

### ISSUE THREAD

**Issue by GreenLunar** [SparkyBluefang/PDMTS#1]
PDMTS does not preserve modification timestamps of downloaded files under version 31.7

**Comment by srx6** [comment]
Neither under 39.0
I am sorry I can't contribute in code analysis. Just reporting.

**Comment by warp9pnt9** [comment]
http://forums.mozillazine.org/viewtopic.php?p=13156229#p13156229

The new downloads API does not provide access to the necessary information, so it will not work with Firefox 26+.

https://bugzilla.mozilla.org/show_bug.cgi?id=918188

**Comment by GreenLunar** [comment]
Honestly, drop this so called web browser. Use decent web browsers.

I use QupZilla.

**Comment by warp9pnt9** [comment]
I don't care what anyone uses and I'm not here to enter browser war debates.  I encountered this problem and found this thread with no answers, and the answer elsewhere, so I put the answer here as well.  That is all.  If, for whatever reason, someone else comes along with the same question, then now they know it's not an AddOn problem, but a browser API "problem".

However, the precise rationale eludes me as to why the API support was dropped, i.e. was it intentional (some misguided attempt to increase security by decreasing functionality), or a mere oversight that none of the Mozilla devs seem to care about or get around to.  If it was simply an oversight, and patches submitted, I wonder if they'd have been implemented by now.  More likely, simple fixes get stuck in 10+ year bickering amongst devs, and any and all attempts to patch get rejected.  But that is conjecture based on suspicion and intuition, not fact.

**Comment by GreenLunar** [comment]
> I don't care what anyone uses and I'm not here to enter browser war debates.

No, this is not a debate, to me.

The corporation responsible for this web browser does not care for our freedoms nor the preferences of its users.  They have deprecated dynamic titles for bookmarks, they are gradually deprecating RSS, and they have foisted DRM on all of us.

They appear to actively ignore free speech.  Do not support them.  Use truly free utilities.

**Comment by warp9pnt9** [comment]
My response was not merely for the OP.  The comment is more for the benefit of others who may have been looking for the reason "why" the addon was not working, and who would possibly end up here as I did (one of the top Google searches), and disappointed to find no specific explanation.  I filled that gap.

This is an issue tracker for an addon in Mozilla Firefox, so I choose to constrain discussion to the addon at hand, within the context of those users who have chosen or continue to choose to use that browser.

Ultimately, it in this context, it is better to refrain from evangelical diatribe about the merits or faults of a browser.  There are many other forums for that.
